test/test_sparse_csr.py

```diff
@@ -1122,8 +1122,77 @@ def test_sparse_csr_unary(self, device, dtype, op):
             assert torch.is_tensor(expected)
             output = op(sample.input.to_sparse_csr())
             assert torch.is_tensor(output)
+
             self.assertEqual(output.to_dense(), expected)
 
+    @ops(sparse_csr_unary_ufuncs)
+    def test_sparse_csr_unary_out(self, device, dtype, op):
+        samples = op.sample_inputs(device, dtype)
+
+        if len(samples) == 0:
+            self.skipTest("Skipped! No sample inputs!")
+
+        if not op.supports_out:
+            self.skipTest("Skipped! Out not supported")
+
+        for sample in samples:
+            assert torch.is_tensor(sample.input)
+            # Sparse CSR only supports 2D tensors as inputs
+            # Fail early to prevent silent success with this test
+            if sample.input.ndim != 2:
+                raise ValueError("Expected 2D tensor but got tensor with dimension: {sample.input.ndim}.")
+
+            sample.input = sample.input.to_sparse_csr()
+            expect = op(sample.input, *sample.args, **sample.kwargs)
+
+            out = self.genSparseCSRTensor(sample.input.size(), sample.input._nnz(),
```

Comment:
In a follow-up, I think we should also test the behavior when `out.shape == (0, 0)`, `out._nnz() == 0` (`out` should be resized correctly in this case), and when `out.shape` is wrong.

Reply:
Acknowledging that I've taken note of this. Thanks, @IvanYashchuk - I'll address this in a follow-up PR :)